app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -0,0 +1,271 @@
+import org.osmdroid.views.MapView
+
+object MapScreen {
+  const val TEST_TAG_MAP = "map"
+  const val TEST_TAG_MENU_BUTTON = "menuButton"
+  const val TEST_TAG_HIKES_LIST = "hikesList"
+  const val TEST_TAG_HIKE_ITEM = "hikeItem"
+}
+
+@Composable
+fun MapScreen(
+    hikingRoutesViewModel: ListOfHikeRoutesViewModel =
+        viewModel(factory = ListOfHikeRoutesViewModel.Factory)
+) {
+  val context = LocalContext.current
+  // Avoid re-creating the MapView on every recomposition
+  val mapView = remember { MapView(context) }
+
+  Configuration.getInstance().apply {
+    // Set user-agent to avoid rejected requests
+    userAgentValue = context.packageName
+
+    // Allow for faster loading of tiles. Default OSMDroid value is 2.
+    tileDownloadThreads = 4
+
+    // Maximum number of tiles that can be downloaded at once. Default is 40.
+    tileDownloadMaxQueueSize = 40
+
+    // Maximum number of bytes that can be used by the tile file system cache. Default is 600MB.
+    tileFileSystemCacheMaxBytes = 600L * 1024L * 1024L
+  }
+
+  mapView.apply {
+    controller.setZoom(15.0)
+    // TODO : Set the center to the user's location
+    controller.setCenter(GeoPoint(46.5, 6.6))
+    // Enable touch-controls such as pinch to zoom
+    setMultiTouchControls(true)
+    // Update hiking routes every time the user moves the map
+    // TODO : Those updates could be quite frequent, have a cooldown to avoid sending a request each
+    // time?
+    addMapListener(
+        object : MapListener {
+          override fun onScroll(event: ScrollEvent?): Boolean {
+            hikingRoutesViewModel.setArea(mapView.boundingBox)
+            return true
+          }
+
+          override fun onZoom(event: ZoomEvent?): Boolean {
+            hikingRoutesViewModel.setArea(mapView.boundingBox)
+            return true
+          }
+        })
+  }
+
+  Box(modifier = Modifier.fillMaxSize()) {
+    AndroidView(
+        factory = { mapView }, modifier = Modifier.fillMaxSize().testTag(MapScreen.TEST_TAG_MAP))
+
+    IconButton(
+        onClick = {
+          // TODO : Adapt the map screen to navigation
+          Toast.makeText(context, "Menu not implemented yet", Toast.LENGTH_SHORT).show()
+        },
+        modifier =
+            Modifier.padding(16.dp)
+                .align(Alignment.TopStart)
+                // Clip needs to be before background
+                .clip(RoundedCornerShape(8.dp))
+                .background(MaterialTheme.colorScheme.surface)
+                .testTag(MapScreen.TEST_TAG_MENU_BUTTON)) {
+          Icon(
+              Icons.Default.Menu,
+              contentDescription =
+                  context.getString(R.string.map_screen_menu_button_content_description),
+              tint = MaterialTheme.colorScheme.onSurface)
+        }
+
+    CollapsibleHikesList(hikingRoutesViewModel)
+  }
+
+  // Initialize the list of hiking routes once when the map is loaded
+  // Leave subsequent updates to the map listener
```

Comment:
Question: What does this mean? Does the Launched Effect take over? I am not sure what this comments means

Reply:
Thanks for clarifying, the updated comment is good!